### ISSUE TITLE
hashcat hangs when called with --left option

### ISSUE THREAD

**Issue by hgbstu** [hashcat/hashcat#893]
It is not possible to get uncracked hashes with the --left option while cracking instance is still running.

**How to reproduce**
try to crack a hash that is not cracked immediately
> 098f6bcd4621d373cade4e832627b4f6:test
> 5fa0fd04d89bbf2f09c891add76ac48c:hashcat-uncrackable$_-_

content of md5.hash
> 098f6bcd4621d373cade4e832627b4f6
> 5fa0fd04d89bbf2f09c891add76ac48c

run hashcat
> hashcat64.bin -m 0 -a 3 md5.hash ?a?a?a?a?a?a?a?a?a?a --potfile-path=result.pot -i --increment-max=10

while hashcat is cracking the hashes run following command
> hashcat64.bin md5.hash -m 0 --potfile-path=result.pot --left

the command doesn't return (hangs) until the cracking instance has finished or is aborted.

**Affected versions**
tested with hashcat-3.20-rc5 (30-Nov-2016 16:53)
hashcat v3.10 is not affected, the --left command returns immediately

**Comment by jsteube** [comment]
This is some effect that is related to the potfile file locking. You can see this through a strace:

```
fcntl(3, F_SETLKW, {l_type=F_WRLCK, l_whence=SEEK_SET, l_start=0, l_len=0}
```
^ blocking there

@magnumripper Do you really think we need the potfile locking? Hashcat could handle this use case out of the box if the locking is not active

**Comment by magnumripper** [comment]
We *absolutely* need the locking. However, we're [supposedly] only locking during actual write so something seems to be amiss. I will look into this ASAP!

**Comment by magnumripper** [comment]
BTW another test would be to run two instances of HC using the same pot file. This is supposed to work just fine, with no noticable performance problem (unless of course both instances crack thousands of passwords per second - that would mean a little performance drop but the pot file will end up correct and not thrashed).

**Comment by hgbstu** [comment]
Tried starting two HC instances writing to same pot file, strace shows the same out as @jsteube mentioned in his post. So the second instance locks.

**Comment by magnumripper** [comment]
Apparently the --show and --left options use potfile_write_open which is a bit unintuitive as it's only supposed to read. My fix in #895 works around it.
